src/targets/gpu/include/migraphx/gpu/where.hpp

```diff
@@ -0,0 +1,33 @@
+#ifndef MIGRAPHX_GUARD_RTGLIB_WHERE_HPP
+#define MIGRAPHX_GUARD_RTGLIB_WHERE_HPP
+
+#include <migraphx/argument.hpp>
+#include <migraphx/reflect.hpp>
+#include <migraphx/op/where.hpp>
+#include <migraphx/gpu/miopen.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace gpu {
+
+struct context;
+
+struct hip_where
+{
+    op::where op;
```

Comment:
This member field makes the type not empty which is why it can't be serialized. However, instead of defining your own class, just use `ternary_device` class:

```cpp
struct hip_where : ternary_device<hip_where, device::where>
{
};
```